### PR TITLE
chore(interfaces): Update interfaces for ICS20, ICS20Authorization and Vesting

### PR DESCRIPTION
**Pull request by Vvaradinov**

### CODE REVIEW

File: precompiles/common/IICS20Authorization.sol [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /// @param allocations An array of Allocation authorized with this grant.
+    event IBCTransferAuthorization(
+        address indexed grantee,
+        address indexed granter,
+        ICS20Allocation[] allocations
+    );
+
+    /// @dev Approves IBC transfer with a specific amount of tokens.
+    /// @param grantee The address for which the transfer authorization is granted.
+    /// @param allocations An array of Allocation for the authorization.
```
Comment (0xstepit) [inline precompiles/common/IICS20Authorization.sol:22]:
```suggestion
    /// @param allocations An array of ICS20Allocation for the authorization.
```

File: precompiles/common/IICS20Authorization.sol [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    function approve(address grantee, ICS20Allocation[] calldata allocations) external returns (bool approved);
+
+    /// @dev Revokes IBC transfer authorization for a specific grantee.
+    /// @param grantee The address for which the transfer authorization will be revoked.
+    function revoke(address grantee) external returns (bool revoked);
+
+    /// @dev Increase the allowance of a given grantee by a specific amount of tokens for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
```
Comment (0xstepit) [inline precompiles/common/IICS20Authorization.sol:30]: Is the contract of an address only or a general address?

File: precompiles/common/IICS20Authorization.sol [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    function increaseAllowance(
+        address grantee,
+        string calldata sourcePort,
+        string calldata sourceChannel,
+        string calldata denom,
+        uint256 amount
+    ) external returns (bool approved);
+
+    /// @dev Decreases the allowance of a given grantee by a specific amount of tokens for IBC transfer methods.
+    /// @param grantee The address of the contract that is allowed to spend the granter's tokens.
```
Comment (0xstepit) [inline precompiles/common/IICS20Authorization.sol:45]: Same as above

Comment (0xstepit) [reply precompiles/common/IICS20Authorization.sol:45]: I cannot delete previous comment (???) it is referred to line 45 not 46

File: precompiles/common/Types.sol [hunk 4]
```diff
@@ -1,5 +1,14 @@
 // SPDX-License-Identifier: LGPL-3.0-only
-pragma solidity >=0.8.17 .0;
+pragma solidity >=0.8.17;
+
+
+/// @dev Allocation represents a single allocation for an IBC fungible token transfer.
```
Comment (0xstepit) [inline precompiles/common/Types.sol:5]:
```suggestion
/// @dev ICS20Allocation represents a single allocation for an IBC fungible token transfer.
```

Comment (0xstepit) [reply precompiles/common/Types.sol:5]: Also, Coin is a list but the doc states "an IBC fungible token transfer"

File: precompiles/stateful/Vesting.sol [hunk 5]
```diff
@@ -25,12 +24,83 @@ struct Period {
...
-/// @dev The interface through which solidity contracts will interact with vesting module.
+/// @dev The interface through which solidity contracts will interact with vesting.
+/// We follow this same interface including four-byte function selectors, in the precompile that
+/// wraps the pallet.
 /// @custom:address 0x0000000000000000000000000000000000000803
 interface VestingI {
+    /// @dev This event is emitted when the allowance of a granter is set by a call to the approve method.
+    /// The value field specifies the new allowance and the methods field holds the information for which methods
+    /// the approval was set.
+    /// @param grantee The contract address that received an Authorization from the granter.
```
Comment (0xstepit) [inline precompiles/stateful/Vesting.sol:35]: Only contract?

**Review by 0xstepit — COMMENTED** [review]
LGTM, great job @Vvaradinov! Left just some comments/questions.

File: precompiles/common/IICS20Authorization.sol (hunk 2 above)
Comment (Vvaradinov) [reply precompiles/common/IICS20Authorization.sol:30]: It's a contract address and takes the origin as the `granter`

**Pull request merged**